### PR TITLE
Using browserSet property.

### PR DESCRIPTION
**Pull request by Blackbaud-BobbyEarl**

### CODE REVIEW

File: vsts/config/utils/browsers.js [hunk 1]
```diff
@@ -4,26 +4,98 @@
...
+    bsBrowserIE,
+    bsBrowserFirefox,
+    bsBrowserSafari
+  ]
+};
+
 // We normalize properties despite Browserstack/Protractor/Karma using different keys.
-const map = {
-  'e2e': {
+// Leaving this functionaly in case we expose the e2e property in the future.
```
Comment (Blackbaud-SteveBrush) [inline vsts/config/utils/browsers.js]: Did you mean "functionality"?

File: vsts/config/utils/browsers.js [hunk 2]
```diff
@@ -4,26 +4,98 @@
...
+  browser: 'Firefox'
+};
+
+const bsBrowserSafari = {
+  os: 'OS X',
+  osVersion: 'High Sierra',
+  browser: 'Safari'
+};
+
+const browserSets = {
```
Comment (Blackbaud-SteveBrush) [inline vsts/config/utils/browsers.js]: This is great; love it.

File: vsts/config/utils/browsers.js [hunk 3]
```diff
@@ -4,26 +4,98 @@
...
+      config,
+      `skyPagesConfig.skyux.testSettings.${testSuite}.browserSet`,
+      ''
+    );
+    const configBrowserSetValidated = validateBrowserSet(configBrowserSetRequsted);
+
+    const allowedPropertiesMap = propertiesMap[testSuite];
     const allowedPropertiesKeys = Object.keys(allowedPropertiesMap);
 
+    // User hasn't defined a browserSet or provided an invalid set.
```
Comment (Blackbaud-SteveBrush) [inline vsts/config/utils/browsers.js]: Should be "User hasn't defined a browserSet, or has provided an invalid set."?

Also, is there a default browser set if the user does not provide one?

File: vsts/config/utils/browsers.js (hunk 1 above)
Comment (Blackbaud-BobbyEarl) [reply vsts/config/utils/browsers.js]: Yep.

File: vsts/config/utils/browsers.js (hunk 3 above)
Comment (Blackbaud-BobbyEarl) [reply vsts/config/utils/browsers.js]: I cleaned up the wording slightly, but I'm not quite following what was wrong with the original.  Let me know what you think about the new one.

Great question regarding a default browser.  If the property isn't set, or is set to something invalid, the local browser still runs.  Setting this property kicks you in to browserstack mode.

Comment (Blackbaud-SteveBrush) [reply vsts/config/utils/browsers.js]: Makes total sense, thanks for the clarification. In regard to the comment, I was reading it as:
"User hasn't defined a browserSet or (hasn't) provided an invalid set." (Basically, carrying the negative "hasn't" to the next section of the sentence.) It's not a huge deal either way :)

Comment (Blackbaud-BobbyEarl) [reply vsts/config/utils/browsers.js]: Ahhh I see.  Yep that's confusing.  I'll make the comment more clear.

**Pull request merged**